### ISSUE TITLE
reproductibily issue

### ISSUE THREAD

**Issue by mouadriyad** [keras-team/keras-tuner#421]
Hi,

I had executed my code two times and I got different results despite fixing the seeds of tensorflow, numpy, python envirement, and the oracle.
I used a custom loop as the snippet code shows.

Did I miss something?

`
...Preambule...

class CVTuner(kerastuner.engine.tuner.Tuner):
  def run_trial(self, trial, x, y, batch_size=32, epochs=1):
    cv = model_selection.KFold(5)
    val_losses = []
    import timeit
    tic=timeit.default_timer()
    for train_indices, test_indices in cv.split(x):
      x_train, x_test = x[train_indices], x[test_indices]
      y_train, y_test = y[train_indices], y[test_indices]
      model = self.hypermodel.build(trial.hyperparameters)
      model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs)
      val_losses.append(model.evaluate(x_test, y_test))
      clear_output()
    toc=timeit.default_timer()
    print('time: '+str(toc-tic))
    self.oracle.update_trial(trial.trial_id, {'val_loss': np.mean(val_losses)})
    self.save_model(trial.trial_id, model)

for i in list:
    tuner = CVTuner(
    hypermodel=EEGNet,directory=('/gdrive/My Drive/app2/Hyperpara/BO_3_'+str(i+1)),
    oracle=kerastuner.oracles.BayesianOptimization(
        objective='val_loss',seed=0,
        max_trials=20))

    X_train    = X_train_set[i][0]
    #X_validate = data[ranges[A],:,:]
    X_test     = X_test_set[i][0]

    Y_train    = Y_train_set[i][0]
    #Y_validate = labels[ranges[A]]
    Y_test     =Y_test_set[i][0]
    X_train    = X_train.astype("float32")
    X_test     = X_test.astype("float32")
    #X_validate = X_validate.astype("float32")

    Y_test     = Y_test.astype("int")
    Y_train    = np_utils.to_categorical(Y_train.astype("int"),4)
    tuner.search(X_train[:,:,64:256*2+64], Y_train, batch_size=128, epochs=400)

    best_hps = tuner.get_best_hyperparameters()[0]
    print(best_hps.values)

`
